gcp-observability/src/main/java/io/grpc/gcp/observability/ObservabilityConfigImpl.java

```diff
@@ -74,95 +87,128 @@ void parse(String config) throws IOException {
-            "'global_trace_sampling_rate' needs to be between [0.0, 1.0]");
-        // Using alwaysSample() instead of probabilitySampler() because according to
-        // {@link io.opencensus.trace.samplers.ProbabilitySampler#shouldSample}
-        // there is a (very) small chance of *not* sampling if probability = 1.00.
-        if (1 - samplingRate < EPSILON) {
-          this.sampler = Samplers.alwaysSample();
-        } else {
-          this.sampler = Samplers.probabilitySampler(samplingRate);
-        }
-      }
-      Map<String, ?> rawCustomTags = JsonUtil.getObject(config, "custom_tags");
-      if (rawCustomTags != null) {
-        ImmutableMap.Builder<String, String> builder = new ImmutableMap.Builder<>();
-        for (Map.Entry<String, ?> entry: rawCustomTags.entrySet()) {
-          checkArgument(
-                  entry.getValue() instanceof String,
-                  "'custom_tags' needs to be a map of <string, string>");
-          builder.put(entry.getKey(), (String) entry.getValue());
-        }
-        customTags = builder.build();
+            entry.getValue() instanceof String,
+            "'labels' needs to be a map of <string, string>");
+        builder.put(entry.getKey(), (String) entry.getValue());
       }
+      customTags = builder.build();
     }
   }
 
-  private EventType convertEventType(String val) {
-    switch (val) {
-      case "GRPC_CALL_UNKNOWN":
-        return EventType.GRPC_CALL_UNKNOWN;
-      case "GRPC_CALL_REQUEST_HEADER":
-        return EventType.GRPC_CALL_REQUEST_HEADER;
-      case "GRPC_CALL_RESPONSE_HEADER":
-        return EventType.GRPC_CALL_RESPONSE_HEADER;
-      case "GRPC_CALL_REQUEST_MESSAGE":
-        return EventType.GRPC_CALL_REQUEST_MESSAGE;
-      case "GRPC_CALL_RESPONSE_MESSAGE":
-        return EventType.GRPC_CALL_RESPONSE_MESSAGE;
-      case "GRPC_CALL_TRAILER":
-        return EventType.GRPC_CALL_TRAILER;
-      case "GRPC_CALL_HALF_CLOSE":
-        return EventType.GRPC_CALL_HALF_CLOSE;
-      case "GRPC_CALL_CANCEL":
-        return EventType.GRPC_CALL_CANCEL;
-      default:
-        throw new IllegalArgumentException("Unknown event type value:" + val);
+  private List<LogFilter> parseRpcEvents(List<?> rpcEvents) {
+    List<Map<String, ?>> jsonRpcEvents = JsonUtil.checkObjectList(rpcEvents);
+    ImmutableList.Builder<LogFilter> rpcEventsListBuilder =
+        new ImmutableList.Builder<>();
+    for (Map<String, ?> jsonClientRpcEvent : jsonRpcEvents) {
+      rpcEventsListBuilder.add(parseJsonLogFilter(jsonClientRpcEvent));
     }
+    return rpcEventsListBuilder.build();
   }
 
+
   private LogFilter parseJsonLogFilter(Map<String, ?> logFilterMap) {
-    return new LogFilter(JsonUtil.getString(logFilterMap, "pattern"),
-        JsonUtil.getNumberAsInteger(logFilterMap, "header_bytes"),
-        JsonUtil.getNumberAsInteger(logFilterMap, "message_bytes"));
+    Boolean exclude = JsonUtil.getBoolean(logFilterMap, "exclude");
+    Boolean global = false;
```

Comment:
See my comment above about using unboxed types. In this case `global` can definitely be primitive `boolean` even if `LogFilter` constructor required the boxed `Boolean` type but that should be changed to primitive type as well.

Even `exclude` can be primitive.
```
boolean exclude = Boolean.valueOf(JsonUtil.getString(logFilterMap, "exclude"));
```

Reply:
Since value for "exclude" is a boolean, `JsonUtil.getString" will throw an exception when it is not string. 
Hence, updated to use `Boolean.TRUE.equals(JsonUtil.getBoolean(logFilterMap, "exclude"))`.

Reply:
Yes, correct. I missed the `"true"` vs `true` difference. Looks good